api/all/src/main/java/io/opentelemetry/api/internal/Utils.java

```diff
@@ -30,4 +32,17 @@ public static void checkArgument(boolean isValid, String errorMessage) {
       throw new IllegalArgumentException(errorMessage);
     }
   }
+
+  /**
+   * Logs a warning message if the argument is false.
+   *
+   * @param logger the logger instance that writes message.
+   * @param isValid whether the argument check passed.
+   * @param warnMessage the message to use for the warning log.
+   */
+  public static void warnOnArgument(Logger logger, boolean isValid, String warnMessage) {
```

Comment:
I saw we just inline this in BatchLogRecordPorcessorBuilder. Its pretty trivial to abstract, and adds additional [shared internal code](https://github.com/open-telemetry/opentelemetry-java/issues/6970), which we're trying to move away from.